gitprovider/resources.go

```diff
@@ -68,6 +68,9 @@ type UserRepository interface {
 	// DeployKeys gives access to manipulating deploy keys to access this specific repository.
 	DeployKeys() DeployKeyClient
 
+	// DeployTokens gives access to manipulating deploy tokens to access this specific repository.
+	DeployTokens() DeployTokenClient
```

Comment:
```suggestion
	DeployTokens() (DeployTokenClient, error)
```
I think we need to return an error type (and a sentinel error) here to signify that a specific implementation might not support deploy tokens.

Reply:
Thanks for the early feedback @makkes 🎉 

I've addressed this and returned `gitprovider.ErrNoProviderSupport` for now. Is that good enough or does it make sense to create a dedicated error?

Reply:
`gitprovider.ErrNoProviderSupport` is exactly right!